trigger.go

```diff
@@ -1,59 +1,106 @@
 package zabbix
 
-import (
-	//"fmt"
-	"github.com/AlekSi/reflector"
-)
-
 type (
+	// SeverityType of a trigger
+	// Zabbix severity see : https://www.zabbix.com/documentation/3.2/manual/api/reference/trigger/object
 	SeverityType int
 )
 
 const (
+	// Different severity see : https://www.zabbix.com/documentation/3.2/manual/config/triggers/severity
+
+	// NotClassified is Not classified severity
 	NotClassified SeverityType = 0
-	Information   SeverityType = 1
-	Warning       SeverityType = 2
-	Average       SeverityType = 3
-	High          SeverityType = 4
-	Critical      SeverityType = 5
+	// Information is Information severity
+	Information SeverityType = 1
+	// Warning is Warning severity
+	Warning SeverityType = 2
+	// Average is Average severity
+	Average SeverityType = 3
+	// High is high severity
+	High SeverityType = 4
+	// Critical is critical severity
+	Critical SeverityType = 5
+)
 
-	Enabled  StatusType = 0
+const (
+	// Enabled trigger status enabled
+	Enabled StatusType = 0
+	// Disabled trigger status disabled
 	Disabled StatusType = 1
+)
+
+const (
+	// Trigger value see : https://www.zabbix.com/documentation/3.2/manual/config/triggers
 
-	OK      ValueType = 0
+	// OK trigger value ok
+	OK ValueType = 0
+	// Problem trigger value probleme
 	Problem ValueType = 1
 )
 
-// https://www.zabbix.com/documentation/2.2/manual/appendix/api/item/definitions
+// TriggerFunction The function objects represents the functions used in the trigger expression
+type TriggerFunction struct {
+	FunctionID string `json:"functionid"`
+	ItemID     string `json:"itemid"`
+	Function   string `json:"function"`
+	Parameter  string `json:"parameter"`
+}
+
+// TriggerFunctions is an array of TriggerFunction
+type TriggerFunctions []TriggerFunction
+
+// Trigger represent Zabbix trigger object
+// https://www.zabbix.com/documentation/3.2/manual/api/reference/trigger/object
 type Trigger struct {
-	TriggerId   string `json:"triggerid,omitempty"`
+	TriggerID   string `json:"triggerid,omitempty"`
 	Description string `json:"description"`
 	Expression  string `json:"expression"`
 	Comments    string `json:"comments"`
 	//TemplateId  string    `json:"templateid"`
-	Value ValueType `json:""`
-
-	Priority SeverityType `json:"priority"`
-	Status   StatusType   `json:"status"`
+	//Value ValueType `json:""`
+
+	Priority     SeverityType     `json:"priority,string"`
+	Status       StatusType       `json:"status,string"`
+	Dependencies Triggers         `json:"dependencies,omitempty"`
+	Functions    TriggerFunctions `json:"functions,omitempty"`
+	// Items contained by the trigger in the items property.
+	ContainedItems Items `json:"items,omitempty"`
+	// Hosts that the trigger belongs to in the hosts property.
+	TriggerParent Hosts `json:"hosts,omitempty"`
```

Comment:
This is documented as:
> Return the hosts that the trigger belongs to in the `hosts` property. 

I'd rather stay close to the Zabbix API naming and name this `ParentHosts` (plural).